src/cli_plugin/install/zip.js

```diff
@@ -82,6 +82,10 @@ export function analyzeArchive(archive) {
             plugins.push({
               id: manifest.id,
               stripPrefix: match[1],
+              folderName:
```

Comment:
Why can't this just be `.manifest.folderName`? There's nothing wrong with a null value, it shouldn't become blank if it's not set. If you move the logic to resolve `targetFolderName` into a function you can check for `null` or blank there.

Reply:
Updated to move the logic to `getTargetFolderName` function.